src/app/clinic-dashboard/general/pre-appointment-outreach/pre-appointment-outreach.component.ts

```diff
@@ -0,0 +1,353 @@
+import { Component, OnInit } from '@angular/core';
+import { ClinicDashboardCacheService } from '../../services/clinic-dashboard-cache.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import * as _ from 'lodash';
+import { Subscription } from 'rxjs';
+import { PreAppointmentOutreachResourceService } from 'src/app/etl-api/pre-appointment-outreach-resource.service';
+
+interface ReportParams {
+  locationUuids: string;
+  yearWeek: string;
+  processOutcome: number;
+}
+
+@Component({
+  selector: 'app-pre-appointment-outreach',
+  templateUrl: './pre-appointment-outreach.component.html',
+  styleUrls: ['./pre-appointment-outreach.component.css']
+})
+export class PreAppointmentOutreachComponent implements OnInit {
+  public errors: any[] = [];
+  public preAppointmentOutreachList: any[] = [];
+  public locationUuids: string;
+  public params: any;
+  public routeSub: Subscription = new Subscription();
+  public loadingpreAppointmentOutreachList = false;
+
+  public weeks: any[] = [];
+  public selectedWeek: string;
+  public selectedFormattedWeek: string;
+
+  // constants
+  private ALL = 'All';
+  private FOLLOW_UP_SUCCESSFUL = 'Follow-up Successful';
+  private FAILED_FOLLOW_UP_ATTEMPT = 'Failed Follow-up Attempt';
```

Comment:
Would it be better if we used enum for the constants e.g
enum Status {
  ALL = 'All',
  FOLLOW_UP_SUCCESSFUL = 'Follow-up Successful',
  FAILED_FOLLOW_UP_ATTEMPT = 'Failed Follow-up Attempt'
}

Reply:
At this point we aren't sure if these the only filter options, that said let's refactor this while doing the enhancements!